daemon/create.go

```diff
@@ -151,5 +153,9 @@ func (daemon *Daemon) VolumeCreate(name, driverName string, opts map[string]stri
 	if err != nil {
 		return nil, err
 	}
+
+	if (driverName != "" && v.DriverName() != driverName) || (driverName == "" && v.DriverName() != volume.DefaultDriverName) {
```

Comment:
@tiborvass  I did that way in the first time. But as @cpuguy83  comments,  "This should go in daemon.VolumeCreate (in daemon/create.go) instead of here.
In cases where you have docker run -v some_name:/foo --volume-driver other_driver and some_name is an existing volume but on a different driver, this should work."